src/component/elements/InputRange.jsx

```diff
@@ -24,135 +24,45 @@ const styles = {
-    oldX = e.clientX;
-    setToZero();
-  } else {
-    oldX = e.clientX;
-    firstCalc = false;
-  }
-}
-
-function setToZero() {
-  clearTimeout(timerId);
-  timerId = setTimeout(() => {
-    speedX = 0;
-  }, 50);
-}
-
-function InputRange({
+export default function InputRange({
   name,
-  minValue,
-  maxValue,
-  value: valueProps,
+  value,
   onChange,
   label,
   style,
   className,
-  noPropagation,
 }) {
-  const mouseEnterHandler = useCallback(() => {
-    if (stateValues.step === 'moving' || stateValues.step === 'start') {
-      stateValues = {
-        ...stateValues,
-        step: stateValues.step === 'start' ? 'initial' : 'end',
-      };
-    }
-  }, []);
-
-  const mouseDownHandler = useCallback(
-    (event) => {
-      if (noPropagation) {
-        event.stopPropagation();
-      }
-
-      const boundingRect = event.currentTarget.getBoundingClientRect();
-      const x = event.clientX - boundingRect.x;
-      const y = event.clientY - boundingRect.y;
-      if (stateValues.step === 'initial' || stateValues.step === 'end') {
-        stateValues = {
-          startX: x,
-          startY: y,
-          startScreenX: event.screenX,
-          startScreenY: event.screenY,
-          boundingRect: event.currentTarget.getBoundingClientRect(),
-          step: 'start',
-          clientX: event.clientX,
-          clientY: event.clientY,
-        };
-      }
-
-      return false;
-    },
-    [noPropagation],
-  );
-
-  const mouseMoveCallback = (event) => {
-    if (stateValues.step === 'start' || stateValues.step === 'moving') {
-      const { screenX, screenY } = event;
-      stateValues = {
-        ...stateValues,
-        step: 'moving',
-        endX: stateValues.startX + screenX - stateValues.startScreenX,
-        endY: stateValues.startY + screenY - stateValues.startScreenY,
-      };
-      calcSpeed(event);
-      const valueRange = maxValue - minValue;
-      const positionDiff = stateValues.endX - stateValues.startX;
-      const value =
-        (Math.abs(positionDiff) * (Math.abs(speedX) > 3 ? valueRange : 5)) /
-        stateValues.boundingRect.width;
+  const previousPosition = useRef(0);
 
+  function mouseMoveCallback(event) {
+    let diff = event.clientX - previousPosition.current;
+    previousPosition.current = event.clientX;
+    if (event.buttons === 1) {
       onChange({
-        value: speedX >= 0 ? valueProps + Math.abs(value) : valueProps - value,
+        value: value + diff / (event.shiftKey ? 10 : 1),
         name,
       });
     }
-  };
+  }
 
-  const mouseUpCallback = () => {
-    if (stateValues.step === 'moving' || stateValues.step === 'start') {
-      stateValues = {
-        ...stateValues,
-        step: stateValues.step === 'start' ? 'initial' : 'end',
-      };
-    }
-  };
+  function mouseUpCallback() {
+    window.removeEventListener('mousemove', mouseMoveCallback);
+    window.removeEventListener('mouseup', mouseUpCallback);
+  }
+
+  function mouseDownCallback() {
```

Comment:
Should this be wrapped in a useEffect in order to be able to remove the event in all the cases ?

Reply:
do you think it's possible to not have the mouseup event?